sdk/src/message.rs

```diff
@@ -178,6 +186,10 @@ impl Sanitize for Message {
             if ci.program_id_index as usize >= self.account_keys.len() {
                 return Err(SanitizeError::IndexOutOfBounds);
             }
+            // A program cannot be a payer.
+            if ci.program_id_index == 0 {
```

Comment:
Does this jive with the first non-loader key as fee-payer in `load_tx_accounts()` ?

Reply:
I think so, I believe it just means that `account_keys[0]` and also `accounts[0]` should always be considered the fee-payer and should not be referenced as a program index.

Technically I think it means we don't need to do the payer_index calculation, but I feel a bit better with it. What do you think?

Reply:
Yep, agreed!  Maybe log a warning in the payer_index calculation if payer_index != 0 ?